velox/docs/develop/scalar-functions.rst

```diff
@@ -538,6 +538,20 @@ The code below shows an example of a function that concatenates a variable numbe
        }
      };
 
+Error Handling
+^^^^^^^^^^^^^^
+
+Simple functions do not need to handle exceptions in the "call" function (or any of
+its variations) or the "initialize" function. If an exception is thrown, the
+framework will ensure it is either propagated to the Task or the semantics of
+Velox's builtin TRY expression are respected if applicable.
```

Comment:
If we are talking about error handling, we should mention the preferred method of exposing error conditions using Status object. In case that is not possible and the implementation has to throw, its allowed to capture exceptions if thrown by an API call inside the implementation in order to wrap in into a more meaningful exception (with possibly additional relevant context).  However, it should ensure it always propagates (or never catches) VeloxRuntimeErrors.